lib/improver/spotdata/extract_data.py

```diff
@@ -227,44 +288,65 @@ def make_cube(self, cube, data, sites):
         non_spatial_dimcoords = cube.dim_coords[0:n_non_spatial_dimcoords]
-        cube.coord('time').convert_units('hours since 1970-01-01 00:00:00')
-
-        latitudes = [float(site['latitude']) for site in sites.itervalues()]
-        longitudes = [float(site['longitude']) for site in sites.itervalues()]
-        altitudes = [np.nan_to_num(site['altitude'])
-                     for site in sites.itervalues()]
-        utc_offsets = [float(site['utc_offset'])
-                       for site in sites.itervalues()]
-        wmo_sites = [site['wmo_site'] for site in sites.itervalues()]
-
-        indices, latitude, longitude, altitude, utc_offset, wmo_site = (
-            self._build_coordinates(
-                latitudes, longitudes, altitudes, utc_offsets, wmo_sites))
-
         dim_coords = [coord for coord in non_spatial_dimcoords]
+
+        # Add an index coordinate as a dimension coordinate.
+        indices = self._build_coordinate(np.arange(len(sites)), 'index',
+                                         data_type=int)
         dim_coords.append(indices)
+
+        # Include the forecast_reference_time from the source cube.
+        try:
+            forecast_ref_time = cube.coord('forecast_reference_time')
+        except CoordinateNotFoundError:
+            raise CoordinateNotFoundError(
+                'No forecast reference time found on source cube.')
+
+        forecast_ref_time.convert_units('seconds since 1970-01-01 00:00:00')
+        forecast_periods = cube.coord('time').points - forecast_ref_time.points
+        forecast_period = self._build_coordinate(
+            forecast_periods, 'forecast_period', units='seconds')
+
+        # Build the auxiliary coordinates.
+        crds = self._aux_coords_to_make()
+        aux_crds = []
+        for key, kwargs in zip(crds.keys(), crds.itervalues()):
+            aux_data = np.array([entry[key] for entry in sites.itervalues()])
+            crd = self._build_coordinate(aux_data, key, **kwargs)
+            aux_crds.append(crd)
+
+        # Construct zipped lists of coordinates and indices.
         n_dim_coords = len(dim_coords)
         dim_coords = zip(dim_coords, range(n_dim_coords))
-        aux_coords = zip([latitude, longitude, altitude, utc_offset, wmo_site],
-                         [n_dim_coords-1]*5)
+        aux_coords = zip(aux_crds, [n_dim_coords-1]*len(aux_crds))
+
+        # Copy other cube metadata.
+        metadata_dict = copy.deepcopy(cube.metadata._asdict())
 
         # Add leading dimension for time.
         data = np.expand_dims(data, axis=0)
         result_cube = Cube(data,
-                           long_name=cube.name(),
                            dim_coords_and_dims=dim_coords,
                            aux_coords_and_dims=aux_coords,
-                           units=cube.units)
+                           **metadata_dict)
+        result_cube.add_aux_coord(forecast_ref_time)
+        result_cube.add_aux_coord(forecast_period, 0)
```

Comment:
The cube can't have a forecast reference time as we have created it from scratch, so this is the point at which we add it.

Good spot about my lazy 0 index assumption; this is only true if I've promoted time to an axis in the first step (which is normally true given the time extraction before this step). I've changed it to grab the time coordinate dimension and use that.